### ISSUE TITLE
Undefined property: Collective\IronQueue\IronQueue::$crypt

### ISSUE THREAD

**Issue by Jmrich** [LaravelCollective/iron-queue#13]
As the title says it is looking for a variable named `crypt` that is never defined. Not sure when this became an issue. Here is the code that fails:

```
public function pushRaw($payload, $queue = null, array $options = [])
    {
        if ($this->shouldEncrypt) {
            $payload = $this->crypt->encrypt($payload);
        }

        return $this->iron->postMessage($this->getQueue($queue), $payload, $options)->id;
    }
```

**Comment by stayallive** [comment]
It looks like in the connecter $crypt is defined but never passed along the the actual IronQueue instance.

> https://github.com/LaravelCollective/iron-queue/blob/041eca66ffb891c990bd6576ca7617d74e363cae/src/Connectors/IronConnector.php#L13-L18

This is very weird since I have always used this package with crypt on for ages... now questioning everything about it 😨

**Comment by lastcow** [comment]
Same issue here, only can do 'encrypt'=>false.

**Comment by carvefx** [comment]
@Jmrich can you please provide a Laravel version and an iron-queue version? I use iron-queue in production with encrypt => true and it's working as expected, so I'm guessing a newer laravel version somehow broke the instantiation?

I actually took a look at other Queue Connectors shipped with laravel, and none seem to support encryption, so the case where laravel simply broke something is actually quite likely

**Comment by Jmrich** [comment]
Laravel version "5.2.41", Iron-queue "v5.2.4". In my production app my laravel is v 5.2.29 with iron-queue at v 5.2.4, and it works fine. So thats the only thing that I can think of. I just noticed this in my dev env when I tried to add something to the queue

**Comment by patrikengborg** [comment]
Same issue here, same version numbers.

**Comment by golonix** [comment]
I've created a fix for this. Inability to send encrypted data is serious drawback.
Authors/maintainers of this package (@tshafer, @adamgoose), please address this issue as soon as possible.

Thanks

**Comment by carvefx** [comment]
bump @tshafer

**Comment by kuwana** [comment]
I don't know reason of no acceptance for #14 .
Anybody can show me other solution?

**Comment by jean-alex** [comment]
Quick fix....
Since Laravel 5.2.4*, the protected property [$crypt](https://github.com/laravel/framework/blob/5.1/src/Illuminate/Queue/Queue.php#L224) of [Queue](https://github.com/laravel/framework/blob/5.2/src/Illuminate/Queue/Queue.php) ( Illuminate\Queue\Queue) ) has been changed to [$encrypter](https://github.com/laravel/framework/blob/5.2/src/Illuminate/Queue/Queue.php#L28). But LaravelCollective (v5.2.4) has not been updated to match that change.

To be able to use LaraCollective/iron-queue, you have two options 
1) Set 'encrypt' to false in your queue config file
2) Update [Collective\IronQueue\IronQueue](https://github.com/LaravelCollective/iron-queue/blob/5.2/src/IronQueue.php) to replace $this->crypt by $this->encrypter at lines [93](https://github.com/LaravelCollective/iron-queue/blob/5.2/src/IronQueue.php#L93) and [238](https://github.com/LaravelCollective/iron-queue/blob/5.2/src/IronQueue.php#L238).

**Comment by golonix** [comment]
@zoomsurhaiti there is pull request that fixes this issue (#14)

**Comment by golonix** [comment]
It seems like original authors/maintainers ditched this package. I've decided to fork it and maintain it on my own. You can find it here: https://github.com/HydrefLab/iron-queue

Feel free to use it or contribute :)

**Comment by trideout** [comment]
Fixes merged into v5.2.5 release
